internal/provider/provider.go

```diff
@@ -186,3 +192,34 @@ func (p *Provider) HandleMountRequest(ctx context.Context, cfg config.Config) (*
 		Files:         files,
 	}, nil
 }
+
+func (p *Provider) GetRotatedSecret(ctx context.Context, itemName string, cfg config.Config) (string, error) {
+	body := akeyless.GetRotatedSecretValue{
+		Names: itemName,
+	}
+	body.SetJson(true)
+	if cfg.UsingUID() {
+		body.SetUidToken(config.GetAuthToken())
+	} else {
+		body.SetToken(config.GetAuthToken())
+	}
+
+	gsvOut, res, err := config.AklClient.GetRotatedSecretValue(ctx).Body(body).Execute()
+	if err != nil {
+		if errors.As(err, &apiErr) {
+			return "", fmt.Errorf("can't get secret value: %v", string(apiErr.Body()))
+		}
+		return "", fmt.Errorf("can't get secret value: %w", err)
+	}
+	defer res.Body.Close()
+	val, ok := gsvOut["value"]
+	if !ok {
+		return "", fmt.Errorf("can't get secret: %v", itemName)
+	}
```

Comment:
can we reuse?

Reply:
The only one call we need for rotated secrets. Not sure I understand where I can reuse it

Reply:
the only thing that changes here is `config.AklClient.GetRotatedSecretValue` no? let's extract it if we can and pass that the function as a value

Reply:
Body type also different, the way we retrieve value from map also ...